### PR TITLE
Remove path from HTTP Client span names

### LINKED ISSUE #515: URLs in HTTP Client Span Names
## Is your feature request related to a problem? Please describe

The HTTP client uses the request URL as part of span names (`GET http://service/api/v1/user/12345/books`)
This results into duplicate span names which only differ in the variable parts (in above example the `:userID`)
The tracing system and it's UI/UX will be affected by an ever growing number of span names.

## Is this a bug? Please provide steps to reproduce, a failing test etc.

Use the Patron HTTP Client to call an API endpoint using variables in the URL path and have tracing enabled.

## Describe the solution

Dropping the request path from the span names.
A PR will be opened to implement the describe solution.

### PR DESCRIPTION
**Pull request by siavashs**
## Which problem is this PR solving?

Closes #515

## Short description of the changes

The HTTP Client uses the complete URl in span names.
It results in duplicate span names with variations in the paths including IDs.

This change uses only the `[scheme]://[host:port]` part of the URL in span names.

Some unused span related functions are also removed.

Signed-off-by: Siavash Safi <siavash.safi@gmail.com>

### CODE REVIEW

File: client/http/http.go [hunk 1]
```diff
@@ -126,28 +125,8 @@ func (tc *TracedClient) do(req *http.Request) (*http.Response, error) {
 	return rsp, nil
 }
 
-func span(path, corID string, r *http.Request) (opentracing.Span, *http.Request) {
```
Comment (mantzas) [inline client/http/http.go:129]: QQ: Was this here unused?

File: client/http/http.go [hunk 2]
```diff
@@ -126,28 +125,8 @@ func (tc *TracedClient) do(req *http.Request) (*http.Response, error) {
...
-func finishSpan(sp opentracing.Span, code int) {
-	ext.HTTPStatusCode.Set(sp, uint16(code))
-	ext.Error.Set(sp, code >= http.StatusInternalServerError)
-	sp.Finish()
-}
-
-func opName(method, path string) string {
-	return method + " " + path
+func opName(method, scheme, host string) string {
+	return method + " " + scheme + "://" + host
```
Comment (mantzas) [inline client/http/http.go:129]: Could we add a test to catch when this one breaks?

File: client/http/http.go (hunk 1 above)
Comment (pkritiotis) [reply client/http/http.go:129]: Yes, I can confirm that this function was not used

File: client/http/http.go (hunk 2 above)
Comment (siavashs) [reply client/http/http.go:129]: Done.

**Review by mantzas — APPROVED** [review]

**Review by pkritiotis — APPROVED** [review]

**Pull request merged**